### PR TITLE
Allow convenient filtering of metric descriptors by type prefix.

### PR DESCRIPTION
**Pull request by rimey**
@supriyagarg Could you please also take a look at this?

### CODE REVIEW

**Comment by supriyagarg** [comment]
This PR looks good to me.

File: system_tests/monitoring.py [hunk 1]
```diff
@@ -79,6 +79,18 @@ def test_list_metric_descriptors(self):
             self.assertTrue(label.value_type)
             self.assertTrue(label.description)
 
+    def test_list_metric_descriptors_filtered(self):
+        client = monitoring.Client()
+
+        PREFIX = 'compute.googleapis.com/'
+        descriptors = client.list_metric_descriptors(type_prefix=PREFIX)
+
+        # There are currently 18 types with this prefix, but that may change.
```
Comment (dhermes) [inline system_tests/monitoring.py]: This comment says 18 but the number below is 10. What am I not understanding?

**Comment by dhermes** [comment]
LGTM. @rimey Am I good to merge this for you?

**Comment by rimey** [comment]
@dhermes Yes, it's good to go.

**Pull request merged**